packages/ingest/test/consumer.js

```diff
@@ -7,56 +7,75 @@ const test = require('ava');
 const { sleep } = require('@cumulus/common/util');
 
 const consumer = rewire('../consumer');
-const Consume = consumer.Consume;
-
-const timeToReceiveMessages = 10;
-let messageLimit = 3;
-class MyTestConsumerClass extends Consume {
-  constructor() {
-    super();
-    this.messageLimit = messageLimit;
-    this.timeLimit = timeToReceiveMessages * 2 + 100;
-  }
-}
-let myTestConsumerClass;
+const Consumer = consumer.Consumer;
+
+const timeToReceiveMessages = 1 * 1000;
+let testConsumer;
 
-async function stubReceiveSQSMessages() {
+async function stubReceiveSQSMessages(_url, { numOfMessages }) {
   await sleep(timeToReceiveMessages);
-  return ['hi', 'bye'];
+  return Array.apply(null, { length: numOfMessages }).map(() => 'i am a message'); // eslint-disable-line prefer-spread
 }
 consumer.__set__('receiveSQSMessages', stubReceiveSQSMessages);
+console.log(consumer.__get__('receiveSQSMessages'));
+consumer.__set__('deleteSQSMessage', async () => true);
 function processFn() {}
 
 const sandbox = sinon.sandbox.create();
 let processSpy;
 
 test.beforeEach(() => {
   // need to reinstantiate because this.now = Date.now()
-  myTestConsumerClass = new MyTestConsumerClass();
-  processSpy = sandbox.spy(myTestConsumerClass, 'processMessage');
+  testConsumer = new Consumer();
+  processSpy = sandbox.spy(testConsumer, 'processMessage');
 });
 test.afterEach.always(() => sandbox.restore());
 
 test.serial('stops after timelimit', async (t) => {
-  const result = await myTestConsumerClass.processMessages(processFn, messageLimit);
-  t.is(result, 4);
-  t.is(processSpy.callCount, 4);
+  testConsumer.messageLimit = 40;
```

Comment:
There's lots of mental math going on in here for me.  I know it's easy math, but could we set these up as something easier to grok?  message limit 40 means a max of 4 batches of 10 to the processMessages routine. 

what if we renamed timeToReceiveMessages => timeToReceive10Messages ?

Reply:
Because it's not always 10 messages. If the message limit ends in a digit other than 0, we make the final call with the remaining limit, i.e. 25 = receive(10), receive(10), receive(5). I'll add another spy to hopefully make it clearer.